apps/web/src/client/components/file-uploader/_html.js

```diff
@@ -1,43 +1,51 @@
+import { errorMessage } from './_errors.js';
+
+/** @typedef {{message: string, fileRowId: string, name: string}} AnError */
+/** @typedef {File & {fileRowId?: string}} FileWithRowId */
+
 /**
  *
  * @param {number} sizesInBytes
  * @returns {string}
  */
 const renderSizeInMainUnit = (sizesInBytes) => {
-	// TODO: not totally clear, for now always returns the size in MB
+	const unit = sizesInBytes > 99_999 ? 'MB' : 'KB';
+	const power = sizesInBytes > 99_999 ? 1e-5 : 1e-2;
 
-	return `${Math.round(sizesInBytes * 1e-5) / 10} MB`;
+	return `${Math.round(sizesInBytes * power) / 10} ${unit}`;
 };
 
 /**
- * @param {File} uploadedFile
- * @returns {string}
+ * @param {FileWithRowId} uploadedFile
+ * @returns {HTMLElement}
  */
 export const buildRegularListItem = (uploadedFile) => {
-	const fileRowId = `file_row_${uploadedFile.lastModified}_${uploadedFile.size}`;
+	const li = document.createElement('li');
```

Comment:
creating the element and then filling it up with the innerHtml is the best approach here because returning a Node rather than a string allows me to use `rows.append()` rather than `rows.innerHTML+=`

they both work, but innerHTML removes all the EventListeners set on its elements, which meanse that two loops are necessary: a first one to insert all the html elements and a second one to set eventListeners.
With .append(NODE) I can inserting new html elements without unsetting the eventlisteners.